packages/ui/src/components/ProgressBar/index.tsx

```diff
@@ -80,19 +90,77 @@ export const ProgressBar = ({
   sentiment = 'primary',
   className,
   'data-testid': dataTestId,
+  showProgress = false,
+  label,
+  labelDescription,
+  direction = 'column',
+  'aria-labelledby': ariaLabelledBy,
+  'aria-label': ariaLabel,
 }: ProgressBarProps) => (
-  <StyledProgressContainer
-    role="progressbar"
-    aria-valuenow={value}
-    aria-valuemin={0}
-    aria-valuemax={100}
-    className={className}
-    data-testid={dataTestId}
+  <Stack
+    gap={direction === 'column' ? 1 : 2}
+    direction={direction}
+    alignItems="center"
   >
-    {progress ? (
-      <StyledProgress />
-    ) : (
-      <StyledFilled sentiment={sentiment} value={value} />
-    )}
-  </StyledProgressContainer>
+    {direction === 'column' && (label || showProgress) ? (
+      <Stack
+        direction="row"
+        justifyContent={!label && showProgress ? 'right' : 'space-between'}
+        width="100%"
+      >
+        <Stack gap={1} direction="row" alignItems="center">
+          {label ? (
+            <Text as="label" variant="bodySmallStrong" sentiment="neutral">
+              {label}
+            </Text>
+          ) : null}
+          {labelDescription ?? null}
+        </Stack>
+        {showProgress ? (
+          <Text
+            as="label"
+            variant="bodySmall"
+            sentiment="neutral"
+            placement="right"
+          >
+            {value} %
+          </Text>
+        ) : null}
+      </Stack>
+    ) : null}
+    {label && direction === 'row' && labelDescription ? (
+      <Stack direction="row" gap={1}>
+        <Text as="label" variant="bodySmallStrong" sentiment="neutral">
+          {label}
+        </Text>
+        {labelDescription}
```

Comment:
you could wrap label description into a text component if `typeof labelDescription === 'string'`